### DISCUSSION TITLE
G7 Mismatch

### DISCUSSION THREAD

**Discussion by Lynd** [NightscoutFoundation/xDrip#2865] [Q&A] [answered]
Greetings.

 I'm trying to connect a G7 device for the first time.   The QR code does not scan, so I entered the 4 digit code on the insertion device as the "transmitter ID" in xdrip, and the transmitter does not seem to connect.  the Bluetooth Link shows "Disconnected" and the Sensor Status shows "Mismatch".   I re-entered the transmitter ID and received the same results.

**Comment by Navid200** [comment, marked as answer]
You say the QR code does not scan.  Please explain what you mean.
Did you follow this?  https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by filip0k** [reply]
I have the same issue. I've tried all the versions from 3rd of May but it just shows Mismatch. 
I've followed the instructions on the link.

**Reply by rudysroost** [reply]
> I have the same issue. I've tried all the versions from 3rd of May but it just shows Mismatch. I've followed the instructions on the link.

Ditto.

 I'll be glad to help in any way I can.

**Reply by Navid200** [reply]
But, the very first post in this thread reads:  "... the QR code does not scan".

Does that refer to the QR code in the guide, or any other?

If you scanned the QR code in the guide with no problem, and you get a mismatch error, can you post a screenshot of the status page/
Do you happen to have used another transmitter with the exact same pairing code recently?

**Reply by filip0k** [reply]
If it helps, I can see:
`No transmitter mac address! `
in the logs

**Reply by filip0k** [reply]
@Navid200 
I didn't have any issues with QR codes. I could easily scan them. 
But after that it doesn't work with Mismatch status.

![image](https://github.com/NightscoutFoundation/xDrip/assets/37311490/b475afa4-1890-436c-ae3b-2aee2cec43f5)
![image](https://github.com/NightscoutFoundation/xDrip/assets/37311490/bd5494d2-bb79-44b6-81ca-cb3e770c0a8a)

**Reply by Navid200** [reply]
Is this the very first time you are trying G7?
If not, did the previous device have the same pairing code?

**Reply by filip0k** [reply]
@Navid200 
This is the first time I'm trying it with xDrip.
No, pairing code was definitely different.

**Reply by Navid200** [reply]
What is the phone brand and model?
Have you uninstalled Dexcom app?
Do you mind making sure the settings are all correct?  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html
Or, do you mind posting a screenshot of Settings -> G5/G6/Dex1 Debug Settings?

**Reply by rudysroost** [reply]
![image](https://github.com/NightscoutFoundation/xDrip/assets/12412982/e2fb6d86-ca8b-4df2-a208-a86e9ea0a58f)
![image](https://github.com/NightscoutFoundation/xDrip/assets/12412982/106fb3b8-d1f7-4bfa-9ab0-bbfd28acb673)
Pixel 7

**Reply by filip0k** [reply]
Phone: Samsung Galaxy A21S
Android version: 12
Dexcom app is uninstalled.
![cc68addf-14a9-4353-821a-250669480d6a](https://github.com/NightscoutFoundation/xDrip/assets/37311490/bec8cac9-b072-4800-8513-8c29d9eb4bf2)
![dd8e6156-f359-4b0e-8005-a5a03066aa97](https://github.com/NightscoutFoundation/xDrip/assets/37311490/66e030c2-d565-4ad3-9a91-e65c79650426)

**Reply by Navid200** [reply]
Is last connected consistently less than 5 minutes?  Or, has it gone over 5 minutes?

**Reply by rudysroost** [reply]
For me, it is often over five minutes.

**Reply by Navid200** [reply]
@rudysroost If last connected goes over 5 minutes when phone is near transmitter, it is very likely an incorrect setting.
Enable special pairing workaround also if you haven't already.  Have you disabled xDrip battery optimization?  Can you set Bluetooth battery setting to unrestricted?

**Reply by rudysroost** [reply]
Mine is finally working, just like the previous sensor it took many hours and non-intuitive steps to make it work. Basically, set xDrip hardware to Dexcom follow, install Dexcom G7 app and connect it to sensor. Then, change xDrip hardware back to Dexcom transmitter. Uninstall Dexcom app and reboot Android phone.

**Reply by oracess** [reply]
I do not think really you solve the problem, you make the same workaround as described in #2902 ! Many users have the same problem and nobody present a solution and so xdrip+ is only secure usable as follower. If you have a solution to pair a dexcom g7 with a Android 13 handy and dexcom direct please tell me, because nobody reacts officially to this problem since G7 is in xdrip+ nightly.

**Comment by jamorham** [comment]
Anyone with problems of it not working make sure to install the latest nightly version. If you're still having problems please quote which xDrip version you are using.

You don't scan the QR code on the transmitter, if there was any confusion about that.

**Reply by oracess** [reply]
@jamorham 
Have the same problem, see #2902 since G7 is in nightly - no pairing via xdrip+ d46b342-2023.06.17 possible. Please help

**Reply by rudysroost** [reply]
I just started a new sensor with the 2023.06.17 build. Again I had to use the workaround I mentioned three weeks ago. Install the Dexcom app to pair and then uninstall. It works, but it's a bit onerous.

**Reply by oracess** [reply]
@rudysroost
Have still the same problem, see #2902 since G7 is in nightly - no pairing via xdrip+ c680a19-2023.08.23 is possible. Do you have a solution for this problem ? After checking my A53 5G at Samsung (correcting a firmware mistake from July Update) all is fine with BT.
Yesterday I got Rev 0012 from Dexcom and the same story is going on. Also I enabled "Ob1G5CollectionService:v,Ob1G5StateMachine:v" for logging extra tags. You can see after enabling extra tags what's going on, but to say somthing I must wait for pairing a new sensor, because the sensor in log was paired with dexcom app.

![logfile2](https://github.com/NightscoutFoundation/xDrip/assets/41123950/b569d412-f07a-4383-be9c-e08983f1c174)


The full log was sent to jamorham; interesting is the line with "keks" belongs this to "authorization failure" when not paired ?
We will hear in 10 days from jamorham when I sent the full log from pairing process. Sorry for my bad english I am from Austria and don't want google translate.

oracess

**Reply by Navid200** [reply]
You guys can keep posting here and ignoring clear instructions given to you already to help troubleshoot this.
There is nothing we can do to resolve this if you ignore developers.

**Reply by rudysroost** [reply]
@Navid200 The only clear instructions I have found are here: https://navid200.github.io/xDrip/docs/Dexcom/G7.html
I have followed those instructions and still must install the Dexcom app to raise a pairing request. Are there other clear instructions posted elsewhere?

**Reply by Navid200** [reply]
Sorry, my mistake.  I will ask for advice.

**Reply by Navid200** [reply]
Actually, if you look at this discussion, you can see that it is tagged as answered.
Please open a new discussion and describe the problem.
It can be a problem if you describe your problem in someone else's thread and they flag it as answered.
So, I will wait for you to describe the problem in a new thread.  If you have done this already, please provide a link to the discussion.  Hopefully you have not tagged it as answered.

**Reply by oracess** [reply]
I followed also , but all was not useful. Navid200 only give unclear instructions. I will wait for answers from jamorham next week and than i will inform you, because the log will be clearer to me. You cannot download the logfile as .csv or textfile, so it is not easy !

**Reply by jamorham** [reply]
@Navid200 we're waiting for logs (with OB1 debugging enabled) next time the user goes through the pairing process and is unable to get a good result.

**Comment by gazhay** [comment]
Version d46b342-2023.06.17
Dexcom One 

New transmitter today - drops connection and requires re-pairing every 15-20 minutes or so. Getting very frustrating.

Often showing "mismatch" error.

**Reply by Navid200** [reply]
This thread is not about Dexcom One.
Please verify your settings: https://navid200.github.io/xDrip/docs/Intermittent.html

If it asks for pairing every 15-20 minutes, very likely you have not enabled special pairing workaround.

If you are sure all your settings are correct, open a new thread and post screenshots of all the three xDrip settings as well as the Android battery optimization settings.
